clash-lib/src/Clash/Normalize/Strategy.hs

```diff
@@ -28,10 +30,15 @@ import Clash.Rewrite.Util
 
 -- | Normalisation transformation
 normalization :: NormRewrite
-normalization = rmDeadcode >-> constantPropagation >-> rmUnusedExpr >-!-> anf >-!-> rmDeadcode >->
-                bindConst >-> letTL >-> evalConst >-!-> cse >-!-> cleanup >->
-                xOptim >-> rmDeadcode >->
-                cleanup >-> recLetRec >-> splitArgs
+normalization =
+  rmDeadcode >-> constantPropagation >-> rmUnusedExpr >-!-> anf >-!-> rmDeadcode >->
+  bindConst >-> letTL >-!->
+#if !EXPERIMENTAL_EVALUATOR
+  evalConst >->
+#endif
```

Comment:
No, I had removed the whole thing and then in a later commit / amend put it back manually

Reply:
Would it be possible to keep the CPP in one place (`Clash.CPP`) and just do the rest with `if then else`? Inline CPP is hard to read / easy to mess up.

Reply:
Hmm, I wouldn't want to start adding stuff to `Clash.CPP` in `clash-prelude` for this. I don't think it's too bad here, maybe at the start of `testsuite/Main.hs` I could define some macros to tidy up the mess there

Reply:
Using CPP directly makes it clear that this is a compile time choice.
You don't have to wonder where the condition of the `if` comes from and if it could change.
And you can conditionally import modules which don't have to even exist when certain flags are set.